modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/query/contributor/ObjectEntryKeywordQueryContributor.java

```diff
@@ -308,25 +322,49 @@ else if (Objects.equals(
 				queryConfig.addHighlightFieldNames(fieldName);
 			}
 		}
-		else if (Objects.equals(objectField.getDBType(), "Date")) {
+		else if (Objects.equals(
+					objectField.getDBType(),
+					ObjectFieldConstants.DB_TYPE_DATE)) {
+
 			_addNumericClause(
 				"nestedFieldArray.value_date", nestedBooleanQuery, objectField,
 				token);
 		}
-		else if (Objects.equals(objectField.getDBType(), "Double")) {
+		else if (Objects.equals(
+					objectField.getDBType(),
+					ObjectFieldConstants.DB_TYPE_DOUBLE)) {
+
 			_addNumericClause(
 				"nestedFieldArray.value_double", nestedBooleanQuery,
 				objectField, token);
 		}
-		else if (Objects.equals(objectField.getDBType(), "Integer")) {
+		else if (Objects.equals(
+					objectField.getDBType(),
+					ObjectFieldConstants.DB_TYPE_INTEGER)) {
+
 			_addNumericClause(
 				"nestedFieldArray.value_integer", nestedBooleanQuery,
 				objectField, token);
 		}
-		else if (Objects.equals(objectField.getDBType(), "Long")) {
-			_addNumericClause(
-				"nestedFieldArray.value_long", nestedBooleanQuery, objectField,
-				token);
+		else if (Objects.equals(
+					objectField.getDBType(),
+					ObjectFieldConstants.DB_TYPE_LONG)) {
+
+			if (Objects.equals(
+					objectField.getBusinessType(),
+					ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT) &&
+				!_isValidInput(token, ObjectFieldConstants.DB_TYPE_LONG)) {
+
+				nestedBooleanQuery.add(
+					new WildcardQueryImpl(
```

Comment:
I'll check with him, thanks!
Regarding your example, the search will only return `this_is_my_file.jpeg` if the search keyword has the initial letters of the filename, for example: `this_is_my` or `this_is` or even `this_is_my_file.jpeg`. I thought this would be better than having to type the entire filename every time.

Reply:
Got it! Thanks for answering, @carolmariaabb.

Reply:
I discussed with Luiz and Julia and the partial search sould only work if `objectField.isIndexedAsKeyword()` is true (already implemented [here](https://github.com/carolmariaabb/liferay-portal/blob/9a3e37e85b9236caccb7f38625cae76e7fd08d37/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/query/contributor/ObjectEntryKeywordQueryContributor.java#L242-L254)), otherwise the user must search using the entire filename.
I'll change this.